client/native/client.go

```diff
@@ -329,6 +329,18 @@ func (c *Client) listen(ctx context.Context, wg *sync.WaitGroup) {
 			continue
 		}
 
+		info, err := c.Asterisk().Info(nil)
+		if err != nil {
+			Logger.Error("failed to get info from Asterisk", "error", err)
+			time.Sleep(time.Second)
+			continue
+		}
+		if c.node != "" && c.node != info.SystemInfo.EntityID {
+			c.node = info.SystemInfo.EntityID
+		}
+		// We are connected again
+		c.connected = true
```

Comment:
This is good to note:  if we have been connected and we had a failure, we never set connected to `true` again.

However, if we set `connected=true` here, we should _remove_ it from the main `Connect()` routine above (line 309), to avoid confusion down the road about who is responsible for setting the connected flag.

Reply:
Removed it from line 309.